cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+FUNCTION(SET_INTERNAL VAR_NAME VALUE)
+    SET(${VAR_NAME} ${VALUE} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(ADD_FLAG_IF_SUPPORTED)
+    SET(options "")
+    SET(oneValueArgs FLAG TEST_VALUE VALUE)
+    SET(multiValueArgs "")
+    CMAKE_PARSE_ARGUMENTS("" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    IF(${_TEST_VALUE})
+        SET(TEST_FLAG "${_FLAG}=${_TEST_VALUE}")
+    ELSEIF(${_VALUE})
+    SET(TEST_FLAG "${_FLAG}=${_VALUE}")
+    ELSE ()
+        SET(TEST_FLAG "${_FLAG}")
+    ENDIF ()
+    
+    CHECK_CXX_COMPILER_FLAG("${TEST_FLAG}" SUPPORTED)
+
+    IF(SUPPORTED)
+        IF(${_VALUE})
+            SET(_FLAG " ${_FLAG}=${_VALUE}")
+        ENDIF ()
+        STRING(APPEND CMAKE_CXX_FLAGS "${_FLAG}")
+    ENDIF ()
+ENDFUNCTION(ADD_FLAG_IF_SUPPORTED)
+
+FUNCTION(CONFIGURE_COMPILER)
+    INCLUDE(CheckCXXCompilerFlag)
+    IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+        IF(ENABLE_WERROR)
+            STRING(JOIN " "
+                CMAKE_CXX_FLAGS
+                ${CMAKE_CXX_FLAGS}
+                "-Werror"
+                "-Wno-inline"
+                "-Wno-error=cast-align"
+                "-Wmissing-declarations"
+            )
+        ELSE ()
+            ADD_FLAG_IF_SUPPORTED(
```

Comment:
This shouldn't be contingent on -Werror, especially not without any justification.

Reply:
```
if ((NOT DISABLE_WERROR) AND GIT_VERSION)
```
This was an original check.
I just simplified it and got rid of incorrect name - `GIT_VERSION`.
Maybe I misunderstood something or made an error.
If so then please keep in mind my intentions and tell me how do you think it should work then.

Reply:
Yes, that's basically the problem — you merged two different statements (is this release code? Should compiler warnings be errors?) into one boolean. In this case we want to disable debug info only for release code, regardless of -Werror state, so it remains a "reproducible build".

Reply:
Fixed